embeddings/embedding/static/magnitude.py

```diff
@@ -0,0 +1,49 @@
+import hashlib
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, cast
+from urllib.parse import urlparse
+
+import appdirs
+import requests
+from numpy import ndarray
+from pymagnitude import Magnitude
+
+from embeddings.embedding.embedding import Embedding
+from embeddings.utils.loggers import get_logger
+
+_logger = get_logger(__name__)
+
+
+@dataclass
+class MagnitudeConfig:
+    model_path: str
+
+
+class MagnitudeEmbedding(Embedding[List[str], ndarray]):
+    def __init__(self, config: MagnitudeConfig) -> None:
+        self.model = Magnitude(config.model_path)
+
+    def embed(self, data: List[str]) -> ndarray:
+        return cast(ndarray, self.model.query(data))
+
+
+class AutoMagnitude:
+    @staticmethod
+    def from_url(url: str) -> MagnitudeEmbedding:
+        cache_dir = appdirs.user_cache_dir("embeddings", "magnitude")
+        parsed_url = urlparse(url)
+        model_file = os.path.join(
+            cache_dir,
+            f"{hashlib.md5(url.encode('utf-8')).hexdigest()}{os.path.basename(parsed_url.path)}",
+        )
+        Path(cache_dir).mkdir(parents=True, exist_ok=True)
```

Comment:
had to drop idea from point 1, because of how magnitude works etc. so config is just a local path for magnitude to load, and auto magnitude creates that path using URL